src/Parser.js

```diff
@@ -534,77 +518,73 @@ Parser.prototype.callFunction = function(name, args, positions) {
  *
  * @param {string} func  "\name" or "\begin{name}"
  * @param {{numArgs:number,numOptionalArgs:number|undefined}} funcData
- * @param {Array} args  list of arguments to which new ones will be pushed
- * @return the position after all arguments have been parsed
+ * @return the array of arguments, with the list of positions as last element
  */
-Parser.prototype.parseArguments = function(pos, func, funcData, args) {
+Parser.prototype.parseArguments = function(func, funcData) {
     var totalArgs = funcData.numArgs + funcData.numOptionalArgs;
     if (totalArgs === 0) {
-        return pos;
+        return [[this.pos]];
```

Comment:
No, `positions` is [defined](https://github.com/gagern/KaTeX/blob/21a26b807cb1c3b411257e9eddd581a7fe48b2b6/src/functions.js#L68-L69) as

> the positions in the overall string of the function and the arguments.

So even without arguments, the position of the function itself is required. This is in line with how `positions` is [initialized](https://github.com/gagern/KaTeX/blob/21a26b807cb1c3b411257e9eddd581a7fe48b2b6/src/Parser.js#L548) in the case of a non-zero argument count. And in any case, this is preserving existing functionality, we shouldn't change anything here as part of the new position handling.

Moving positions into e.g. parse nodes and the likes will be greatly simplified with all these changes here, I think.